Luft/Settings/LuftSettings.m

```diff
@@ -0,0 +1,119 @@
+static NSString *const kWarningColor = @"luft.warningColor";
+static NSString *const kBadColor = @"luft.badColor";
+static NSString *const kOnlyViewControllers = @"luft.onlyViewControllers";
+static NSString *const kLowerLimit = @"luft.lowerLimit";
+static NSString *const kUpperLimit = @"luft.upperLimit";
+
+@implementation LuftSettings
+
++ (LuftSettings *)sharedSettings {
+    static dispatch_once_t once;
+    static LuftSettings *sharedSettings;
+    dispatch_once(&once, ^ {
+        sharedSettings = [[LuftSettings alloc] init];
+        NSDictionary *defaults = @{kOnlyViewControllers: @YES};
+        [[NSUserDefaults standardUserDefaults] registerDefaults:defaults];
+    });
+    return sharedSettings;
+}
+
+#pragma mark - Setters
+
+- (void)setGoodColor:(NSColor *)color {
+    [self _setColor:color forKey:kGoodColor];
+}
+
+- (void)setWarningColor:(NSColor *)color {
+    [self _setColor:color forKey:kWarningColor];
+}
+
+- (void)setBadColor:(NSColor *)color {
+    [self _setColor:color forKey:kBadColor];
+}
+
+- (void)setOnlyViewControllers:(BOOL)value {
+    [[NSUserDefaults standardUserDefaults] setBool:value forKey:kOnlyViewControllers];
+    [[NSUserDefaults standardUserDefaults] synchronize];
+}
+
+- (void)setLowerLimit:(NSUInteger)limit {
+    [[NSUserDefaults standardUserDefaults] setInteger:limit forKey:kLowerLimit];
+    [[NSUserDefaults standardUserDefaults] synchronize];
+}
+
+- (void)setUpperLimit:(NSUInteger)limit {
+    [[NSUserDefaults standardUserDefaults] setInteger:limit forKey:kUpperLimit];
+    [[NSUserDefaults standardUserDefaults] synchronize];
+}
+
+#pragma mark - Getters
+
+- (NSColor *)goodColor {
+    NSColor *color = [self _getColorForKey:kGoodColor];
+    return color ? color : [self _defaultGoodColor];
+}
+
+- (NSColor *)warningColor {
+    NSColor *color =  [self _getColorForKey:kWarningColor];
+    return color ? color : [self _defaultGoodColor];
+}
+
+- (NSColor *)badColor {
+    NSColor *color = [self _getColorForKey:kBadColor];
+    return color ? color : [self _defaultBadColor];
+}
+
+- (BOOL)onlyViewController {
+    return [[NSUserDefaults standardUserDefaults] boolForKey:kOnlyViewControllers];
+}
+
+- (NSUInteger)lowerLimit {
+    return [[NSUserDefaults standardUserDefaults] integerForKey:kLowerLimit];
+}
+
+- (NSUInteger)upperLimit {
+    return [[NSUserDefaults standardUserDefaults] integerForKey:kUpperLimit];
+}
+
+#pragma mark - Private
+
+- (void)_setColor:(NSColor *)color forKey:(NSString *)key {
+    NSData *colorData = [NSArchiver archivedDataWithRootObject:color];
+    [[NSUserDefaults standardUserDefaults] setObject:colorData forKey:key];
+    [[NSUserDefaults standardUserDefaults] synchronize];
+}
+
+- (NSColor *)_getColorForKey:(NSString *)key {
+    NSData *colorData = [[NSUserDefaults standardUserDefaults] objectForKey:key];
+    if(colorData){
+        return [NSUnarchiver unarchiveObjectWithData:colorData];
+    }
+
+    return nil;
+}
+
+
+- (NSColor *)_defaultGoodColor {
```

Comment:
Theses colors were picker because they work well with my current theme (Solarized Dark). We need to check if they work well for the default theme